### ISSUE TITLE
DruidCoordinatorCleanupOvershadowed will be failed because of IllegalArgumentException

### ISSUE THREAD

**Issue by licl2014** [apache/incubator-druid#5381]
There are many redundant segments  in our cluster,  they are not removed by DruidCoordinatorCleanupOvershadowed  thread and i found coordinator has as follows exception.

```
2018-01-30T14:00:51,750 ERROR [Coordinator-Exec--0] io.druid.server.coordinator.DruidCoordinator:error:97 - Caught exception, ignoring so that schedule keeps going.: {class=io.druid.server.coordinator.DruidCoordinator, exceptionType=class java.lang.IllegalArgumentException, exceptionMessage=Cannot compare against something that is not a LinearPartitionChunk.}
java.lang.IllegalArgumentException: Cannot compare against something that is not a LinearPartitionChunk.
        at io.druid.timeline.partition.LinearPartitionChunk.compareTo(LinearPartitionChunk.java:86) ~[druid-common-0.9.2.jar:0.9.2]
        at io.druid.timeline.partition.LinearPartitionChunk.compareTo(LinearPartitionChunk.java:26) ~[druid-common-0.9.2.jar:0.9.2]
        at java.util.TreeMap.put(TreeMap.java:568) ~[?:1.8.0_101]
        at java.util.TreeSet.add(TreeSet.java:255) ~[?:1.8.0_101]
        at io.druid.timeline.partition.PartitionHolder.add(PartitionHolder.java:62) ~[druid-common-0.9.2.jar:0.9.2]
        at io.druid.timeline.VersionedIntervalTimeline.add(VersionedIntervalTimeline.java:104) ~[druid-common-0.9.2.jar:0.9.2]
        at io.druid.server.coordinator.helper.DruidCoordinatorCleanupOvershadowed.run(DruidCoordinatorCleanupOvershadowed.java:71) ~[druid-server-0.9.2.jar:0.9.2]
        at io.druid.server.coordinator.DruidCoordinator$CoordinatorRunnable.run(DruidCoordinator.java:703) [druid-server-0.9.2.jar:0.9.2]
        at io.druid.server.coordinator.DruidCoordinator$5.call(DruidCoordinator.java:585) [druid-server-0.9.2.jar:0.9.2]
        at io.druid.server.coordinator.DruidCoordinator$5.call(DruidCoordinator.java:578) [druid-server-0.9.2.jar:0.9.2]
        at com.metamx.common.concurrent.ScheduledExecutors$2.run(ScheduledExecutors.java:99) [java-util-0.27.10.jar:?]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_101]
        at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_101]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180) [?:1.8.0_101]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293) [?:1.8.0_101]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [?:1.8.0_101]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [?:1.8.0_101]
```
then i review the code and find DruidCoordinatorCleanupOvershadowed thread do not catch any exception when it cleans up segments. so whether or not catch exception could solve this problem here? @gianm @leventov please have a look.

**Comment by licl2014** [comment]
my druid version is 0.9.2

**Comment by leventov** [comment]
This is likely to be fixed in #3584 which is shipped in Druid 0.10.0, closing this issue unless reproduced on the freshest Druid version.
